File: objects/SO3/testB/testSO3weights.cpp
#include "GElib_base.cpp"
#include "SO3vecB.hpp"
#include "GElibSession.hpp"
#include "SO3vecB.hpp"
#include "SO3weights.hpp"


using namespace cnine;
using namespace GElib;

typedef CtensorB ctensor;


int main(int argc, char** argv){
  GElibSession session(4);
  cout<<endl;

  SO3type tau1({1,2,1});
  SO3type tau2({2,2,3});

  SO3weights W=SO3weights::zero(tau1,tau2);
  cout<<W<<endl;

  SO3vecB u=SO3vecB::gaussian(1,tau1);
  SO3vecB v=u*W;

  cout<<v<<endl;


}

File: python/tests/testSO3weights.py
import torch
import gelib as g

tau1=g.SO3type({0:2,1:3})
tau2=g.SO3type({0:3,1:4})
print(tau1)
print(tau2)

v=g.SO3vec.randn(1,tau1)
print(v)

W=g.SO3weights.randn(tau1,tau2)
print(W)

u=v*W
print(u)


